### ISSUE TITLE
Operator precedence is not taken into account when parsing expressions

### ISSUE THREAD

**Issue by jhrcek** [stil4m/elm-syntax#41]
It seems to me, that operator precedence is not take into account when parsing expressions.

For example the following two expressions
```elm
1 ^ 2 * 3 + 4
1 + 2 * 3 ^ 4
```

both produce the same shape of the expression tree, the first one being obviously incorrect

```
^
│
├─ 1
│
└─ *
   │
   ├─ 2
   │
   └─ +
      │
      ├─ 3
      │
      └─ 4

+
│
├─ 1
│
└─ *
   │
   ├─ 2
   │
   └─ ^
      │
      ├─ 3
      │
      └─ 4
```

I created SSCCE which demonstrated this issue: https://github.com/jhrcek/elm-syntax-sscce

Wider context: this library is used by [elm-review](https://github.com/jfmengels/elm-review), which -among other things - allows writing custom linting rules for elm source code. Unfortunately since the expressions are not parsed correctly by this library, this leads to all sorts of false positives as in the following example

![Screenshot from 2020-04-24 09-34-06](https://user-images.githubusercontent.com/2716069/80186747-cb7f4080-860e-11ea-984c-3d0fd7cb66c7.png)

Because elm-syntax parses this as `(f <| x) || y` it gives this false positive warning, which shouldn't be there, because `||` has higher precedence than `<|`, so the correct parsing should look like `f <| (x || y)`.

**Comment by MartinSStewart** [comment]
@stil4m I think this issue can be closed now.

**Comment by stil4m** [comment]
@jhrcek Thank you for your report. That was clearly a bug.
The fix @MartinSStewart made a functional fix for this which is already released in 7.1.2.

I've just prepared a pull request (https://github.com/stil4m/elm-syntax/pull/66) that will ship with 8.0.0, which will ensure that most programs have proper parsing of operators (and will nog require parsing of dependencies).

**Comment by jhrcek** [comment]
Thank you both @MartinSStewart and @stil4m , I appreciate your work :heart:
